src/interceptors/ClientRequest/NodeClientRequest.ts

```diff
@@ -312,6 +312,17 @@ export class NodeClientRequest extends ClientRequest {
     return this
   }
 
+  flushHeaders(): void {
+    /**
+     * @note `.flusHeaders()` isn't strictly equivalent to `.end()`.
+     * Flushing headers writes request headers but not the request body,
+     * and we likely cannot replace it with `.end()`.
+     *
+     * @see https://github.com/nodejs/node/blob/c2cd74453e7d2794ad81cab63e68371e08bad04f/lib/_http_outgoing.js#L1161
+     */
+    this.end('')
```

Comment:
I don't think this is correct (see the comment above).

We don't distinguish between writing headers and body in the interceptor, we only tap into the `write()` and `end()` methods.